k2/python/tests/array_test.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+#
+# Copyright (c)  2020  Xiaomi Corporation (author: Haowen Qiu)
+#
+# See ../../../LICENSE for clarification regarding multiple authors
+
+# To run this single test, use
+#
+#  ctest --verbose -R array_test_py
+#
+
+import unittest
+
+import torch
+
+import k2
+
+
+class TestFsa(unittest.TestCase):
+
+    def test_int_array2(self):
+        data = torch.arange(10).to(torch.int32)
+        indexes = torch.tensor([0, 2, 5, 6, 10]).to(torch.int32)
+        self.assertEqual(data.numel(),indexes[-1].item())
+
+        array = k2.IntArray2(indexes, data)
+        self.assertFalse(array.empty())
+        self.assertIsInstance(array, k2.IntArray2)
+
+        # test iterator
+        for i, v in enumerate(array):
+            self.assertEqual(i, v)
+
+        self.assertEqual(indexes.numel(), array.size1 + 1)
+        self.assertEqual(data.numel(), array.size2)
+
+        # the underlying memory is shared between k2 and torch;
+        # so change one will change another
+        data[0] = 100
+        self.assertEqual(array.data(0), 100)
```

Comment:
actually, you are unable to use
```python
array.data(0) = 1000
```
to change the `data` in torch in the current implementation.

Reply:
Yes, I know that. That's why I add a `TODO` to support `[]`. BTW, I even doubt whether or not we need to write data in Python, as we will create Array2 with `tensors` or `Array2Size`  and will write data to it in C++

Reply:
@danpovey maybe you can comment on this: do we need to provide `setter` function for `Fsa/Array2` (`indexes` and `data`) in Python code? I just thought it does not make sense (and maybe dangerous) if user can change data in Python directly instead of calling function in C++.

Reply:
Mm.  We'll need some method to create these things, but perhaps you have created one already?
I can see instances where we might need to set the data, at least.